src/model/preferences.ts

```diff
@@ -99,7 +99,7 @@ export async function setPreference<K extends keyof PreferencesDataMap>(
       success: data?.success,
     });
 
-    if (!data.data) {
+    if (!data.success) {
```

Comment:
sometimes data.data is actually null so this was throwing improperly

Reply:
Can the top-level `data` be undefined/null? It's optional chained in the lines above.